website/static/js/fileRevisions.js

```diff
@@ -21,10 +21,22 @@ var Revision = function(data, index, file, node) {
         options.branch = urlParams.branch;
     }
     options[self.versionIdentifier] = self.version;
-    // Note: Google Drive version identifiers often begin with the same sequence
-    self.displayVersion = file.provider === 'googledrive' ?
-        self.version.substring(self.version.length - 8) :
-        self.version.substring(0, 8);
+    self.displayVersion = null;
+    switch (file.provider) {
+        // Note: Google Drive version identifiers often begin with the same sequence
+        case 'googledrive':
+            self.displayVersion = self.version.substring(self.version.length - 8);
+            break;
+        // Note: Dataverse internal version names are ugly; Substitute our own
+        case 'dataverse':
+            self.displayVersion = self.version === 'latest' ? 'Draft' :
```

Comment:
It might be clearer to use a mapping of internal value => display name

```js
var displayMap = {
    'latest': 'Draft',
    'latest-pubished': 'Published
}
```